src/job.rs

```diff
@@ -703,16 +703,15 @@ impl Job {
                 };
                 match chat.typ {
                     Chattype::Group | Chattype::Mailinglist => {
-                        // The next lines are actually what we do in
-                        let (test_normal_chat_id, test_normal_chat_id_blocked) =
-                            chat::lookup_by_contact_id(context, msg.from_id)
-                                .await
-                                .unwrap_or_default();
-
-                        if !test_normal_chat_id.is_unset()
-                            && test_normal_chat_id_blocked == Blocked::Not
-                        {
-                            chat.id.unblock(context).await;
+                        match chat::lookup_by_contact_id(context, msg.from_id).await {
+                            Err(e) => {
+                                warn!(context, "can't get chat: {:#}", e);
+                                return Status::RetryLater;
```

Comment:
i think, this is wrong.

`lookup_by_contact_id()` returns an error if no chat exists for a contact id (i've written a test for that to be sure, #2207) - however, in contrast to the other errors, this is expected and totally fine and the look should continue. the only thing is not to call `unblock()`. this is at least how i understood the original code.

Reply:
indeed, fixed.